### PR TITLE
Download archive files when using BwtB

### PR DESCRIPTION
**Pull request by brentleyjones**
This fixes breakpoints when only the linked binary would be downloaded.

### CODE REVIEW

**Review by erikkerber — APPROVED** [review]
I don't have a great way to reproduce it, but I'll let you know if devs that typically see this see it work more reliably

File: xcodeproj/internal/bazel_integration_files/generate_bazel_dependencies.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ cd "$SRCROOT"
 
 # Calculate Bazel `--output_groups`
 
-readonly swift_outputs_regex='.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
+readonly base_outputs_regex='.*\.a|.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
```
Comment (thii) [inline xcodeproj/internal/bazel_integration_files/generate_bazel_dependencies.sh:9]: Looks like it is missing a `$` sign, but even after adding it the .a files still didn't get downloaded for me.

Comment (thii) [inline xcodeproj/internal/bazel_integration_files/generate_bazel_dependencies.sh:9]:
```suggestion
readonly base_outputs_regex='.*\.a$|.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
```

File: xcodeproj/internal/bazel_integration_files/generate_index_build_bazel_dependencies.sh [hunk 2]
```diff
@@ -12,11 +12,11 @@ readonly output_groups=(
   "bg $BAZEL_TARGET_ID"
 )
 
-readonly swift_outputs_regex='.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
+readonly base_outputs_regex='.*\.a|.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
```
Comment (thii) [inline xcodeproj/internal/bazel_integration_files/generate_index_build_bazel_dependencies.sh:15]:
```suggestion
readonly base_outputs_regex='.*\.a$|.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
```

File: xcodeproj/internal/runner.template.sh [hunk 3]
```diff
@@ -138,13 +138,13 @@ else
   post_config_flags=("${cmd_args[@]:1}")
   if [[ $cmd == "build" && -n "${generator_output_groups:-}" ]]; then
     # `--experimental_remote_download_regex`
-    readonly swift_outputs_regex='.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
+    readonly base_outputs_regex='.*\.a|.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
```
Comment (thii) [inline xcodeproj/internal/runner.template.sh:141]:
```suggestion
    readonly base_outputs_regex='.*\.a$|.*\.swiftdoc$|.*\.swiftmodule$|.*\.swiftsourceinfo$'
```

**Pull request merged**